chain/pool/src/lib.rs

```diff
@@ -512,4 +536,25 @@ mod tests {
         }
         assert_eq!(pool.transaction_size(), 0);
     }
+
+    #[test]
+    fn test_transaction_pool_size_limit() {
+        let pool_size_limit = 1000;
+        let mut pool = TransactionPool::new(TEST_SEED, Some(pool_size_limit));
+        let transactions = generate_transactions("alice.near", "alice.near", 1, 100);
+        let mut total_transaction_size = 0;
+        // Repeatedly fills the pool to capacity and then removes all transactions and starts
+        // again.
+        for tx in transactions.clone() {
```

Comment:
This is an opaque piece of logic. Not clear if both branches of the condition will be hit.
Are generated transactions the same size?
If yes, would it be possible to run the loop `pool_size / tx_size` times checking that it was successfully added, and then one more time to check that it was rejected?

I'm worried that complicated tests sometimes need debugging, and don't simply point to a problem in the code being tested.

Reply:
I agree with this philosophy. Rewrote the test in a way that it is guaranteed to hit a limit on a last transaction.